src/service-store/make-getters.ts

```diff
@@ -78,8 +85,12 @@ export function makeGetters(options: ServiceOptions): ServiceGetters {
           values = values.slice(0, filters.$limit)
         }
 
+        // keep transformed items by $select separately
```

Comment:
I think we shouldn't can  addOrUpdate in the getter as it will cause side effects and that is generally discouraged.
Posva himself have also insisted on that point [there](https://github.com/posva/pinia/discussions/530#discussioncomment-877407
)
I'm currently doing a rather big refactoring and once the store getters, actions and state are all together in the same object, we  get back state types and all and typescrypt complain about that because actions are normally not available on the `this` type in getters.

https://github.com/posva/pinia/discussions/709#discussioncomment-1449092

There isn't any clear way of splitting stores by state, getters and actions yet:
https://github.com/posva/pinia/issues/802#issuecomment-971390377
https://github.com/posva/pinia/issues/829#issue-1062161077

Reply:
It's just for SSR and was discussed before. It makes perfectly sense for SSR. https://feathers-pinia.pages.dev/guide/service-stores.html#automatic-instance-hydration
We need to find a way that it works with `$select`. The downside of my current solution is, that it doesn't return a Model instance which is not reflected by the types

Reply:
Then shouldn't we do it only when we are in an SSR environment ?
https://pinia.esm.dev/ssr/#state-hydration

Reply:
I actually implemented it with full knowledge of the violation of this cardinal sin.  :)  Since the possible negative impact is very low, I consider the API to be acceptable. The possible impact is only one additional render for the entire list.  In order for it to be a problem, the user would need to loop through the results and call a model instance method immediately upon render.  It's a low likelihood that somebody will create an app that requires that.... BUT we require a solution for that scenario.

The location that you pointed out is actually a cleaner implementation, but potentially less performant for apps that pre-load a large number of records.  Converting a large number of records into instances can cause a noticeable delay in loading data for Vue 2.  In Vue 3 I imagine it's going to be barely noticeable.  The existing setup is basically the lazy hydration implementation, if a bit dirty.

So basically, I don't plan on removing the call to `addOrUpdate`, but we do require a solution for those who would prefer a cleaner setup.  A utility to hydrate store data into instances during pinia hydration (https://pinia.esm.dev/ssr/#state-hydration) would be perfect.